libraries/standard/mqtt/src/mqtt_lightweight.c

```diff
@@ -370,11 +373,38 @@ MQTTStatus_t MQTT_SerializePublishHeader( const MQTTPublishInfo_t * const pPubli
     return MQTTSuccess;
 }
 
-MQTTStatus_t MQTT_SerializeDisconnect( const MQTTFixedBuffer_t * const pBuffer )
+MQTTStatus_t MQTT_GetDisconnectPacketSize( size_t * pPacketSize )
 {
+    /* MQTT DISCONNECT packets always have the same size. */
+    *pPacketSize = MQTT_DISCONNECT_PACKET_SIZE;
+
     return MQTTSuccess;
 }
 
+MQTTStatus_t MQTT_SerializeDisconnect( const MQTTFixedBuffer_t * const pBuffer )
+{
+    MQTTStatus_t status = MQTTSuccess;
+    size_t disconnectPacketSize;
+
+    status = MQTT_GetDisconnectPacketSize( &disconnectPacketSize );
+
+    if( status == MQTTSuccess )
+    {
+        if( pBuffer->size < disconnectPacketSize )
+        {
+            status = MQTTNoMemory;
+        }
+    }
```

Comment:
Should this be checked outside of this function after line 169? We are already finding the packet size there, wht we are checking it here again ?